### PR TITLE
Textfile collector: collect files from multiple paths

### PR DESCRIPTION
**Pull request by DiniFarb**
Regarding #1236 and more

Added Features:
- collect files from multiple paths
- walkDir func for all paths to collect files from subdirs
- new flag command `collector.textfile.directories`

About the `collector.textfile.directories` commmand:
For now it is possible to use `collector.textfile.directories`, `collector.textfile.directory` or both. They have the same functionality and in case both commands would be used, the dirs are concatenated. I am not sure if that is ok so lets discuss 😉

As soon all is settled, I'll update the docs - `collector.textfile.md` 😊

### CODE REVIEW

File: collector/textfile.go [hunk 1]
```diff
@@ -250,109 +261,116 @@ func (cr carriageReturnFilteringReader) Read(p []byte) (int, error) {
...
+				return nil
 			}
-			if mf.Help == nil {
-				help := fmt.Sprintf("Metric read from %s", path)
-				mf.Help = &help
+			if !dirEntry.IsDir() && strings.HasSuffix(dirEntry.Name(), ".prom") {
+				_ = level.Debug(c.logger).Log("msg", fmt.Sprintf("Processing file: %s", path))
+				families_array, err := scrapeFile(path, c.logger)
+				if err != nil {
+					_ = level.Error(c.logger).Log("msg", fmt.Sprintf("Error screaping file: %q. Skip File.", path), "err", err)
```
Comment (breed808) [inline collector/textfile.go:283]: `screaping` -> `scraping`

**Review by breed808 — COMMENTED** [review]
Thanks for submitting this!

Is the plan to eventually deprecate the `collector.textfile.directory` flag in favour of the `collector.textfile.directories` flag? If so, it'd be worth adding a deprecation warning to the old flag (see the [process collector](https://github.com/prometheus-community/windows_exporter/blob/b0a1a38829fc5c8052ea41f9e414fe0dea7c9e47/collector/process.go#L80) for an example).

**Comment by DiniFarb** [comment]
I wasn't quite sure about deprecating the `collector.textfile.directory` flag. Technically both flags could stay but it could lead to confusion. Maybe @JDA88 or @LaurenceChau want to give also their input about what flag would be best?  

I think the best way to go is as mentioned from @breed808 - adding a deprecation warning to `collector.textfile.directory` flag and then remove it a few versions later.

**Comment by JDA88** [comment]
> I think the best way to go is as mentioned from @breed808 - adding a deprecation warning to `collector.textfile.directory` flag and then remove it a few versions later.

I agree, as long as the current `--collector.textfile.directory` continue to work like today for at least a few versions

File: collector/textfile.go [hunk 2]
```diff
@@ -63,17 +64,27 @@ type textFileCollector struct {
 func newTextFileCollectorFlags(app *kingpin.Application) {
 	textFileDirectory = app.Flag(
 		FlagTextFileDirectory,
-		"Directory to read text files with metrics from.",
-	).Default(getDefaultPath()).String()
+		"DEPRECATED: Use --collector.textfile.directories",
+	).Default("").String()
```
Comment (breed808) [inline collector/textfile.go:68]: I feel we should add `.Hidden()` here to discourage use of the flag for new users, similar to the deprecated flags in the `process` and other exporters.

**Review by breed808 — COMMENTED** [review]
Looking good! I have one minor concern for the deprecated flag, but everything else is fine.
Thanks for updating this.

EDIT: It'd also be worth rebasing the feature branch on `master` to remove the merge commits.

**Comment by DiniFarb** [comment]
I think this is ready to merge or does it need some further checks/reviews ?

**Review by breed808 — APPROVED** [review]
Looks good! Thanks for your efforts with this one.

**Pull request merged**